main-command/src/main/scala/sbt/internal/client/NetworkClient.scala

```diff
@@ -328,13 +328,39 @@ class NetworkClient(
             term.isSupershellEnabled
           ).mkString(",")
 
-        val cmd = arguments.sbtLaunchJar match {
+        val (cmd: List[String], envs: Map[String, String]) = arguments.sbtLaunchJar match {
           case Some(lj) =>
-            List("java") ++ arguments.sbtArguments ++
-              List("-jar", lj, DashDashDetachStdio, DashDashServer)
+            def loadConfigFile(file: File): List[String] =
+              IO.readLines(file)
+                .map(_.trim)
+                .filterNot(_.startsWith("#"))
+                .filter(_.nonEmpty)
+
+            val jvmOpts =
+              loadConfigFile(file(".jvmopts"))
+            val sbtOpts =
+              loadConfigFile(file("/etc/sbt/sbtopts")) ++ loadConfigFile(file(".sbtopts"))
+
+            val javaOpts = jvmOpts ++ sbtOpts.collect {
```

Comment:
it looks like the .jvmopts are passed both as CLI args and in `JAVA_HOME` in the script, not sure why (`JAVA_HOME` should be enough?) but I replicated that behavior
* https://github.com/sbt/sbt/blob/e4231ac03903e174bc9975ee00d34064a1d1f373/sbt#L739
* https://github.com/sbt/sbt/blob/e4231ac03903e174bc9975ee00d34064a1d1f373/sbt#L746

Reply:
Looking at the 2 snippets above, I realized there is one difference in this implementation compared to the script: the `JAVA_OPTS` parameters are not explicitly passed as CLI `java` args here but they are in the script. Again, I don't think that has any effect, unless we are dealing with a JVM that does not honor `JAVA_OPTS`? But in that case, is it sound to turn the flags there into CLI args anyway?